### PR TITLE
Wiring up AAD token providers to messaging client

### PR DESCRIPTION
**Pull request by serkantkaraca**
This following list is used to make sure that common guidelines for a pull request are followed.

- **Read the [contribution guidelines](./CONTRIBUTING.md).** 
- Make title of the pull request clear and informative. 
- Link to appropriate [issue](https://github.com/Azure/azure-service-bus-dotnet/issues).
- Include test coverage for the changes.

### CODE REVIEW

File: test/Microsoft.Azure.ServiceBus.UnitTests/TokenProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        [DisplayTestMethodName]
+        async Task UseITokenProviderWithAad()
+        {
+            var tenantId = "";
+            var aadAppId = "";
+            var aadAppSecret = "";
+
+            if (string.IsNullOrEmpty(tenantId))
+            {
+                TestUtility.Log($"Skipping test during scheduled runs.");
```
Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/TokenProviderTests.cs]: Would consider adding more context - "Manual test requires authentication information. Skipping during automated runs.".

Comment (SeanFeldman) [reply test/Microsoft.Azure.ServiceBus.UnitTests/TokenProviderTests.cs]: On a second thought, that info is available as environment variables:
https://github.com/Azure/azure-service-bus-dotnet/blob/c58b25b4e242ae5744d690a345fb0b1d77428076/build/build.ps1#L19-L21

Comment (serkantkaraca) [reply test/Microsoft.Azure.ServiceBus.UnitTests/TokenProviderTests.cs]: We can automate the tests later. This requires creation of app identity and secrets as a ARM resource. We are a little in hurry to release the client tomorrow.

**Comment by serkantkaraca** [comment]
Will create another PR on new branch.

File: test/Microsoft.Azure.ServiceBus.UnitTests/TokenProviderTests.cs (hunk 1 above)
Comment (SeanFeldman) [reply test/Microsoft.Azure.ServiceBus.UnitTests/TokenProviderTests.cs]: @serkantkaraca that's exactly what I was implying. `appId` and `clientSecret` are created using ARM resource and stored in the environment for each build execution.

**Comment by SeanFeldman** [comment]
In case this PR is abandoned, could you please remove remote branch `AadTokenProvider`?

**Pull request closed without merging**